server/src/main/java/org/apache/cassandra/sidecar/modules/SchedulingModule.java

```diff
@@ -69,8 +73,24 @@ public PeriodicTaskExecutor periodicTaskExecutor(Vertx vertx,
 
     @ProvidesIntoMap
     @KeyClassMapKey(PeriodicTaskMapKeys.KeyStoreCheckPeriodicTaskKey.class)
-    PeriodicTask keyStoreCheckPeriodicTask(Vertx vertx, Provider<Server> server, SidecarConfiguration configuration)
+    PeriodicTask keyStoreCheckPeriodicTask(Vertx vertx, Provider<Server> server, SidecarConfiguration configuration, WebClientOptions webClientOptions)
     {
-        return new KeyStoreCheckPeriodicTask(vertx, server, configuration);
+        Function<Long, Future<Boolean>> updateServerSSLOptionsFunction =
+        lastModifiedTime -> server.get().updateSSLOptions(lastModifiedTime).compose(v -> Future.succeededFuture(true));
+
+        return new KeyStoreCheckPeriodicTask(vertx, configuration.sslConfiguration(), updateServerSSLOptionsFunction, "ServerKeyStoreCheckPeriodicTask");
```

Comment:
is the plan to use the utility method here?
```suggestion
        return KeyStoreCheckPeriodicTask.forServer(vertx, configuration.sslConfiguration(), updateServerSSLOptionsFunction);
```